File: code/cpp/ch14/practice/practice_02_subarray_sum_k.cpp
/*
 * Practice 2: Subarray Sum Equals K (Count)
 * Chapter 14: Prefix Sums — The Running Total Trick
 *
 * PROBLEM: Count subarrays with sum equal to k.
 *
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */

#include <algorithm>
#include <climits>
#include <iostream>
#include <unordered_map>
#include <vector>
using namespace std;

int solve(vector<int> arr, int k) {
    // TODO: Replace this with your solution
    return 0;
}

// -- Do not change anything below this line --------------------------
int main() {
    int n, k;
    cin >> n;
    vector<int> arr(n);
    for (int i = 0; i < n; i++) cin >> arr[i];
    cin >> k;
    cout << solve(arr, k) << endl;
    return 0;
}

File: code/python/ch14/practice/practice_02_subarray_sum_k.py
"""
Practice 2: Subarray Sum Equals K (Count)
==========================================
Chapter 14: Prefix Sums — The Power of Pre-computation

PROBLEM
-------
Given an array of integers and an integer k, return the total number
of contiguous subarrays whose sum equals k.

INPUT FORMAT
------------
First line: space-separated integers (the array).
Second line: a single integer k.

OUTPUT FORMAT
-------------
A single integer — the count of subarrays with sum equal to k.

CONSTRAINTS
-----------
- 1 <= len(arr) <= 10^5
- -10^6 <= arr[i] <= 10^6
- -10^9 <= k <= 10^9

EXAMPLES
--------
Input:
  1 1 1
  2
Output: 2

Input:
  1 2 3
  3
Output: 2

INSTRUCTIONS
------------
Replace the `pass` in the solve() function with your solution.
The main block at the bottom handles input/output — don't change it.
"""


def solve(arr: list[int], k: int) -> int:
    """Return count of subarrays with sum equal to k."""
    pass  # TODO: Replace this with your solution



# ── Do not change anything below this line ──────────────────────────
if __name__ == "__main__":
    arr = list(map(int, input().split()))
    k = int(input())
    print(solve(arr, k))

